File: results/java/CHECK_CHARACTERS_GIVEN_STRING_CAN_REARRANGED_FORM_PALINDROME_1.cpp
bool f_filled ( string str ) {
  vector < char > vec ;
  for ( int i = 0 ;
  i < str . length ( ) ;
  i ++ ) {
    if ( vec . find ( str [ i ] ) != string :: npos ) {
      vec . erase ( ( vec . find ( str [ i ] ) ) . first ) ;
    }
    else {
      vec . push_back ( str [ i ] ) ;
    }
  }
  if ( str . length ( ) % 2 == 0 && vec . empty ( ) || ( str . length ( ) % 2 == 1 && vec . size ( ) == 1 ) ) {
    return true ;
  }
  else {
    return false ;
  }
}

File: results/java_codex/CHECK_CHARACTERS_GIVEN_STRING_CAN_REARRANGED_FORM_PALINDROME_1.java
bool canFormPalindrome ( string str ) {
  vector < char > list;
  for ( int i = 0;
  i < str . length ( );
  i ++ ) {
    if ( find ( list . begin ( ), list . end ( ), str [ i ] ) != list . end ( ) ) list . erase ( find ( list . begin ( ), list . end ( ), str [ i ] ) );
    else list . push_back ( str [ i ] );
  }
  if ( str . length ( ) % 2 == 0 && list . empty ( ) || ( str . length ( ) % 2 == 1 && list . size ( ) == 1 ) ) return true;
  else return false;
}